pkg/graph/resolvers/system_intake_grb_review_form.go

```diff
@@ -112,3 +112,27 @@ func UpdateSystemIntakeGRBReviewFormInputTimeframeAsync(
 		SystemIntake: updatedIntake,
 	}, nil
 }
+
+// CalcSystemIntakeGRBReviewAsyncStatus calculates the status of the GRB Review Async page
+func CalcSystemIntakeGRBReviewAsyncStatus(
+	intake *models.SystemIntake,
+) *models.SystemIntakeGRBReviewAsyncStatusType {
+	currentTime := time.Now()
+
+	if intake.GrbReviewType != models.SystemIntakeGRBReviewTypeAsync {
+		return nil
+	}
+
+	if intake.GrbReviewAsyncEndDate == nil {
+		return nil
+	}
+
+	// Evaluate if the current time is before the Grb Review Async end date
+	if intake.GrbReviewAsyncEndDate.After(currentTime) {
```

Comment:
```suggestion
	if intake.GrbReviewAsyncEndDate.After(time.Now()) {
```

nit - can declare `currentTime` lower down or even just remove and inline

Reply:
Definitely a valid arguable point. This is an intentional stylistic choice you'll see as a pattern with my code. In accordance with clean code principles, it's often good to cache small variables early in a code block. If it was a significant memory allocation you might prefer RAII, but modern optimizing compilers will typically internally restructure a single-use cached variable to be inline anyway. The idea here is to promote code which can be generally refactored by separation of logic and variables. Not that I really intend to refactor this logic at the moment; it has become fairly habitual and is innocuous or even periodically helps with unexpected downstream cost.